packages/db-base/lib/tools.js

```diff
@@ -15,32 +16,40 @@ function getControllerDir() {
             /* not found */
         }
     }
+
     // Apparently, checking vs null/undefined may miss the odd case of controllerPath being ""
     // Thus we check for falsyness, which includes failing on an empty path
     if (!controllerPath) {
-        const checkPath = path.normalize(path.join(__dirname, '../..'));
-        const pathParts = checkPath.split(path.sep);
-        while (pathParts.length) {
-            const tryPath = pathParts.join(path.sep);
-            if (fs.existsSync(path.join(tryPath, 'lib/tools.js'))) {
-                controllerPath = tryPath;
-                break;
+        let checkPath = path.join(__dirname, '../..');
+        // Also check in the current check dir (along with iobroker.js-controller subdirs)
+        possibilities.unshift('');
+        outer: while (true) {
+            for (const pkg of possibilities) {
+                try {
+                    const possiblePath = path.join(checkPath, pkg);
+                    if (fs.existsSync(path.join(possiblePath, 'lib/tools.js'))) {
+                        controllerPath = possiblePath;
+                        break outer;
+                    }
+                } catch (_a) {
+                    // not found, continue with next possiblity
+                }
             }
-            // Mainly for local development cases
-            if (fs.existsSync(path.join(tryPath, 'iobroker.js-controller/lib/tools.js'))) {
-                controllerPath = path.join(tryPath, 'iobroker.js-controller');
-                break;
-            }
-            if (fs.existsSync(path.join(tryPath, 'ioBroker.js-controller/lib/tools.js'))) {
-                controllerPath = path.join(tryPath, 'ioBroker.js-controller');
+
+            // Controller not found here - go to the parent dir
+            const newPath = path.dirname(checkPath);
+            if (newPath === checkPath) {
+                // We already reached the root dir, abort
                 break;
             }
-            pathParts.pop();
+            checkPath = newPath;
         }
+        // ??? What is this supposed to do ???
         if (controllerPath && !fs.existsSync(controllerPath)) {
             controllerPath = null;
         }
```

Comment:
What is that supposed to do? If js-controller was not found, controllerPath is already null.

Reply:
my IDE told me that null is not allowed for fs.existsSync :-)

Reply:
And in fact it is to secure that the require-resolve (or what ever other logic was used) is really existing ... honestly ... should never occur to be false

Reply:
> null is not allowed for fs.existsSync :-)

That is true, but I didn't get why the entire if branch is there :) This is the check above:
```js
            const possiblePath = require.resolve(pkg);
            if (fs.existsSync(possiblePath)) {
                controllerPath = possiblePath;
```
So if require.resolve found something AND the file exists, then controllerPath is defined. Otherwise it is already `null`.

Reply:
right